src/Components/EditWarehouse/EditWarehouse.jsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from "react";
+import "./EditWarehouse.scss";
+import "../DynamicForm/DynamicForm.scss";
+import DynamicButton from "../DynamicButton/DynamicButton";
+import DynamicForm from "../DynamicForm/DynamicForm";
+import DynamicFormInput from "../DynamicForm/DynamicFormInput";
+import { faArrowLeft } from "@fortawesome/free-solid-svg-icons";
+import PageWrapper from "../PageWrapper/PageWrapper";
+import { useParams } from "react-router-dom";
+import axios from "axios";
+
+const EditWarehouse = ({ handleClick }) => {
+  const { id } = useParams();
+  const [formData, setFormData] = useState({
+    warehouse_name: "",
+    address: "",
+    city: "",
+    country: "",
+    contact_name: "",
+    contact_position: "",
+    contact_phone: "",
+    contact_email: "",
+  });
+
+  const [errors, setErrors] = useState({});
+
+  const validateForm = () => {
+    let newErrors = {};
+
+    if (!formData.warehouse_name)
+      newErrors.warehouse_name = "Warehouse name is required";
+    if (!formData.address) newErrors.address = "Address is required";
+    if (!formData.city) newErrors.city = "City is required";
+    if (!formData.country) newErrors.country = "Country is required";
+    if (!formData.contact_name)
+      newErrors.contact_name = "Contact name is required";
+    if (!formData.contact_position)
+      newErrors.contact_position = "Contact position is required";
+    if (
+      !formData.contact_phone ||
+      !/^[+]?[0-9\s-().]{7,15}$/.test(formData.contact_phone)
+    )
+      newErrors.contact_phone = "Please enter a valid phone number";
+    if (!formData.contact_email || !/\S+@\S+\.\S+/.test(formData.contact_email))
+      newErrors.contact_email = "Please enter a valid email";
+
+    setErrors(newErrors);
+    return Object.keys(newErrors).length === 0;
+  };
+
+  const handleChange = (name, value) => {
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const handleSubmit = async () => {
+    if (validateForm()) {
+      try {
+        const response = await axios.put(
+          `http://localhost:3000/api/warehouses/${id}`,
+          formData
+        );
+
+        if (response.status === 200) {
+          alert("Warehouse has been updated successfully");
+        } else {
+          alert("Error updating warehouse: " + response.data.message);
```

Comment:
Your error message should be specific to what input has the error and also show the red border for that input

in this example u can see the it says 
Please correct the errors in the form  
but the error is with the phone number 
![Screenshot (85)](https://github.com/user-attachments/assets/5f4b03c5-8650-457a-89c6-d42837297c69)